symmetric-tree.cpp

```diff
@@ -0,0 +1,70 @@
+// Author: Ceren Yaşar
+// Question Link: https://leetcode.com/problems/symmetric-tree/
+
+// Time Complexity: O(N)
+// Memory Complexity: O(N)
+
+// This solution uses breadth first search since we start comparisons from the closest child.
+
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+ * };
+ */
+
+class Solution {
+public:
+    bool isSymmetric(TreeNode* root) {
+        
+        // This queue keeps pairs of tree nodes that are supposed to have equal values
+        queue<pair<TreeNode*, TreeNode*>> nextToCheck;
```

Comment:
Altough your solution is good . It is possible to use recursive method.That might not change the complexity of code but provides us shorter code.